### ISSUE TITLE
Experimental conditions autocomplete doesn't behave consistently

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#1846]
(Requested by @CuzickA)

The conditions picker doesn't always work in a consistent way, such that:

- keyboard presses only register under certain conditions, and
- text that the user enters isn't always converted to a tag.

Here's a list of the specific issues:

## Text doesn't convert to tag

This bug can be triggered by first typing a fragment of a term that triggers the autocomplete drop-down (for example, 'high'):

![condition_autocomplete](https://user-images.githubusercontent.com/37659591/56963569-2ba0eb00-6b51-11e9-834d-6e274b28facf.png)

Then, if you lose focus on the text input box (by clicking elsewhere), the text doesn't automatically get converted into a tag:

![autocomplete_no_tag_convert](https://user-images.githubusercontent.com/37659591/56963570-2e034500-6b51-11e9-90a2-7edd1aa329e9.png)

This means that the condition won't actually be saved when you close the popup, which also has the effect of preventing the user from 'finishing' the edit if they haven't changed any other fields &ndash; even though it looks like they have.

The text still converts to a tag when you regain focus (click back onto) the text input box.

**Note** that this automatic conversion on loss of focus (the expected behaviour) _does_ happen if you enter any text that _doesn't_ trigger the autocomplete drop-down (meaning, any term that isn't in the condition list, such as nonsense like 'xxyyzz').

## Keyboard actions are sometimes ignored

Normally, when you press <kbd>Tab</kbd> or <kbd>Enter</kbd> when the drop-down _isn't_ open, the text gets converted to a tag:

![autocomplete_tab_action](https://user-images.githubusercontent.com/37659591/56964062-50e22900-6b52-11e9-9e87-ac40c333a269.png)

However, when the autocomplete is open, <kbd>Tab</kbd> and <kbd>Enter</kbd> don't do anything.

I think that <kbd>Tab</kbd> should focus the first entry on the drop-down list, like what currently happens when you hit the down arrow key <kbd>&darr;</kbd>:

![autocomplete_focus](https://user-images.githubusercontent.com/37659591/56964353-ef6e8a00-6b52-11e9-93da-038dea9df177.png)

An alternative (probably more conventional) is that <kbd>Tab</kbd> should focus the next input on the page; currently, it doesn't even do that.

As for what <kbd>Enter</kbd> should do, that's debatable. I think there are two possible options:

* **Option 1:** Autocomplete by picking the first item of the list. In the example above, this would create a tag with 'high pH'.

* **Option 2:** Create a tag from exactly what the user typed before pressing <kbd>Enter</kbd>. In the example above, this would create a tag with 'high'. This is more consistent with what Google and most search engines do, and it's also consistent with how the autocomplete behaves when the drop-down isn't open.

- - -

I think part of the usability problem could be solved by adding some more text to clarify that the user can complete conditions with <kbd>Tab</kbd> or <kbd>Enter</kbd>, but we have to make sure that actually works in all situations first.

**Comment by jseager7** [comment]
The relevant sources for this are the `conditionPicker` directive in `canto_modules.js`, and `tag-it.js`, which looks like an extension to JQuery UI. That might make the handling of keyboard events difficult to change.

**Comment by ValWood** [comment]
We need an additional line,  has confused people if they start typing something which is not in the existing conditions it isn't clear that you need to hit 'enter' to make the condition appear.

If  you see no suggestions in the autocomplete, suggest a new condition and press 'enter'

**Comment by ValWood** [comment]
@mah11 does this text sound OK.

**Comment by jseager7** [comment]
This issue is blocked by #2025 for now, since this text doesn't describe how the autocomplete actually works:

> If you see no suggestions in the autocomplete, suggest a new condition and press 'enter'

'Enter' doesn't actually do anything in this case, due to a bug described in the issue above.

**Comment by jseager7** [comment]
Half of this issue is fixed by #2158, so I'm moving the other half to a new issue (#2161) and closing this one.

Also, I've changed my mind about changing what the autocomplete hotkeys do, since that could cause problems for any curators who are used to how they already work.
